src/partition/mod.rs

```diff
@@ -72,21 +99,38 @@ impl<'a> SumFilteredPartitionIterator<'a> {
                 }
             }
         }
+
+        // Since all sub-cases of the previous conditional return
+        // unconditionally, the left set matches the filter.
         let next = match self.right_partitions_iterator {
             Some(ref mut iter) => iter.next(),
             None => None,
         };
         match next {
+            // Since the nested partition iterator shares the same filter is
+            // this one, all parts of the partition of the complement (right)
+            // set match, as does the subset (left set). Extend the partition
+            // with the left set as a part, and return the resulting partition
+            // of the initial set.
```

Comment:
```suggestion
            // Since the nested partition iterator shares the same filter as
            // this one, all parts of the partition of the complement (right)
            // set match, as does the subset (left set). Extend the partition
            // with the left set as a part, and return the resulting partition
            // of the initial set.
```